app/components/progress-bar/component.js

```diff
@@ -39,6 +28,11 @@ class ProgressBarComponent extends Component {
    */
   color = null;
 
+  constructor(...args) {
+    super(...args);
+    this.progress = this.args.progress || 0;
+  }
+
```

Comment:
Why do you want to clone the state (`progress`) in the component? Just pass it with a getter for example:
```js
get progress() {
  return this.args.progress || 0;
}
```